pyinterprod/uniprot/unirule.py

```diff
@@ -707,3 +707,216 @@ def update_signatures(filepath: str, uri: str):
     con.commit()
     cur.close()
     con.close()
+
+
+MAX_DOM_BY_GROUP = 20
+DOM_OVERLAP_THRESHOLD = 0.3
+REPR_DOM_DATABASES = ["H", "J", "M", "R", "N"]
+
+
+def get_repr_domains(ora_url: str, output: str = "repr_domains.tsv"):
+    con = oracledb.connect(ora_url)
+    cur = con.cursor()
+
+    placeholders = ', '.join(':' + str(i + 1) for i in range(len(REPR_DOM_DATABASES)))
+    params = {str(i + 1): db for i, db in enumerate(REPR_DOM_DATABASES)}
+    cur.execute(
+        f"""
+        SELECT PROTEIN_AC, H.METHOD_AC, H.DBCODE, POS_FROM, POS_TO, FRAGMENTS
+        FROM INTERPRO.MATCH H
+        INNER JOIN INTERPRO.METHOD D
+        ON H.METHOD_AC = D.METHOD_AC
+        WHERE H.DBCODE in ({placeholders})
+        AND (D.SIG_TYPE = 'D' OR D.SIG_TYPE = 'R')
+        ORDER BY PROTEIN_AC
+        """, params
+    )
+   
```

Comment:
PEP-8 + use a position bind by directly using `REPR_DOM_DATABASES` (it's a list, so the elements of the list will be used as bind values):

```suggestion
    placeholders = ','.join(':' + str(i + 1)
                            for i in range(len(REPR_DOM_DATABASES)))
    cur.execute(
        f"""
        SELECT PROTEIN_AC, H.METHOD_AC, H.DBCODE, POS_FROM, POS_TO, FRAGMENTS
        FROM INTERPRO.MATCH H
        INNER JOIN INTERPRO.METHOD D
        ON H.METHOD_AC = D.METHOD_AC
        WHERE H.DBCODE in ({placeholders})
        AND (D.SIG_TYPE = 'D' OR D.SIG_TYPE = 'R')
        ORDER BY PROTEIN_AC
        """,
        REPR_DOM_DATABASES
    )

```